### PR TITLE
set-to-jail-from-inactive

### PR DESCRIPTION
**Pull request by sasurobert**
set to jailed when account is inactive and with infinite unstaked epoch after jailing is enabled.

### CODE REVIEW

File: process/peer/process.go [hunk 1]
```diff
@@ -661,6 +657,24 @@ func (vs *validatorStatistics) ResetValidatorStatisticsAtNewEpoch(vInfos map[uin
...
+		return
+	}
+
+	if validator.List == string(core.JailedList) && peerAccount.GetList() != string(core.JailedList) {
+		peerAccount.SetListAndIndex(validator.ShardId, string(core.JailedList), validator.Index)
+		return
+	}
+
+	if peerAccount.GetUnStakedEpoch() == core.DefaultUnstakedEpoch && peerAccount.GetList() == string(core.InactiveList) {
+		peerAccount.SetListAndIndex(validator.ShardId, string(core.JailedList), validator.Index)
```
Comment (SebastianMarian) [inline process/peer/process.go:674]: The same set here and in line 669. Why not extract both conditions in shouldBeJailed variable and reduce the code

File: process/peer/process.go [hunk 2]
```diff
@@ -661,6 +657,24 @@ func (vs *validatorStatistics) ResetValidatorStatisticsAtNewEpoch(vInfos map[uin
...
+	if !vs.flagJailedEnabled.IsSet() {
+		return
+	}
+
+	if validator.List == string(core.JailedList) && peerAccount.GetList() != string(core.JailedList) {
+		peerAccount.SetListAndIndex(validator.ShardId, string(core.JailedList), validator.Index)
+		return
+	}
+
+	if peerAccount.GetUnStakedEpoch() == core.DefaultUnstakedEpoch && peerAccount.GetList() == string(core.InactiveList) {
```
Comment (raduchis) [inline process/peer/process.go:673]: no need for the validator.List == string (core.JailedList) ? Could it be not jailed in validator.List and still want to set it to jailed based on the new condition?

**Review by raduchis — APPROVED** [review]

File: process/peer/process.go (hunk 2 above)
Comment (sasurobert) [reply process/peer/process.go:673]: This is for the scenario when in old code it was set to inactive, and should have been set to jailed

File: process/peer/process.go (hunk 1 above)
Comment (sasurobert) [reply process/peer/process.go:674]: Not exactly the same. Would leave as it is

Comment (sasurobert) [reply process/peer/process.go:674]: It is only 1 line of code.

**Review by SebastianMarian — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
System tests passed.

**Pull request merged**